ivy/stateful/helpers.py

```diff
@@ -298,10 +309,11 @@ def _rebuild(self):
         self.build(*self._args, **self._kwargs)
 
     def _compute_module_dict(self):
-        self._module_dict = Container()
```

Comment:
I think this might end up being a problem if we delete a module from a model, because that module would continue to be in the `_module_dict` even if we did `del model.module` or `model.module = new_module`, ensuring it's empty everytime we compute the dict (which is everytime we build). 
What do you think?